src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -370,10 +370,21 @@ private LanguageExpression ConvertAccessExpression(AccessExpression expression,
             }
             else if (resource is DeclaredResourceMetadata declaredResource)
             {
-                // For symbolic-named resources we have the option of simplifying codegen by emitting expressions like "resourceInfo('symbolicName').id".
-                // However, there are numerous cases where resourceInfo can & can't be used, and it's too difficult to try and address them all here.
-                // See https://github.com/Azure/bicep/issues/9450 & https://github.com/Azure/bicep/issues/9246 for examples.
-                // For now, let's stick with emitting the more verbose expressions that we know work.
+                if (context.SemanticModel.Features.ResourceInfoCodegenEnabled)
+                {
+                    // Use simplified "resourceInfo" code generation.
+                    switch (propertyName)
+                    {
+                        case "id":
+                        case "name":
```

Comment:
I'm having trouble finding the original issue about this, but `.name` in Bicep today [will resolve to whatever was supplied for the `name` property](https://github.com/Azure/bicep/pull/16045/files#diff-2b45242bdb455c6c663fcc2de94bbd82a0b1fce22c98dc5e8b0c7ce19d937808R398), whereas `resourceInfo(...).name` will return the fully qualified name. These won't be the same for cases where a template uses nested resources or specifies an explicit parent.

Reply:
Great catch, thank you for pointing this out. Some ideas for solving this without a breaking change:
1. Fall back to existing code gen for nested/parent property resources.
1. Codegen something like the following for nested/parent property resources: `last(split(resourceInfo(...).name), '/')`.
1. Backend change to add an `unqualifiedName` property to the `resourceInfo` function output.

Option 1 works, but is infeasible as it doesn't solve the problem of being able to fully move to `resourceInfo`. Option 2 should work, but feels a little hacky.

My suggestion would be option 2 for now (as it is a client-only change), while we implement option 3, and eventually migrate to using it. Thoughts?

Reply:
That makes sense to me! I would honestly be fine leaving option 2 in place for the long term, though; that looks less complex than a lot of the expressions Bicep generates.

Reply:
Pushed a commit for this